File: PGS/Lv.2/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool comp(const string &a, const string &b) {
	return stoi(a + b) > stoi(b + a);
}

string solution(vector<int> numbers) {
	string answer = "";

	vector<string> v;
	for (auto num : numbers) {
		v.push_back(to_string(num));
	}

	int zero = count(numbers.begin(), numbers.end(), 0);
	if (zero == numbers.size()) {
		answer = '0';
	}
	else {
		sort(v.begin(), v.end(), comp);

		for (auto i : v) {
			answer += i;
		}
	}

	return answer;
}

File: PGS/Lv.2/JAVA/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        
        String[] arr = new String[numbers.length];
        for(int i = 0; i < numbers.length; i++) {
            arr[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(arr, (o1, o2) -> {
            return (o2 + o1).compareTo(o1 + o2);
        });
        
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < arr.length; i++) {
            sb.append(arr[i]);
        }
        
        String answer = sb.toString();
        
        return answer.charAt(0) == '0' ? "0" : answer;
    }
}